src/fparser/common/readfortran.py

```diff
@@ -838,10 +880,30 @@ def _next(self, ignore_comments=None):
                    and ';' in item.get_line():
                 # ;-separator not recognized in pyf-mode
                 items = []
-                for line in item.get_line().split(';'):
+                # Deal with each Fortran statement separately.
+                split_line_iter = iter(item.get_line().split(';'))
+                first = next(split_line_iter)
+                # The first statement has already been processed in
+                # 'item' (and may have label and/or name properties)
+                # but currently has additional invalid content after
```

Comment:
It's probably obvious but how do we know that the additional content is invalid?

Reply:
The 'item' Line object contains the string for the full line e.g. "a=0; b=0; c=0", but we only want "a=0" for the first statement. We can't just throw away the content of 'item' and create a new Line (as we do for subsequent statements), as it may contain a label and/or a construct name which will have already been removed from the original text, so would be lost. An alternative would be to recreate the full line and process again if that were considered a better option. I've updated the text to try to make this more clear in any case.

Reply:
Great, that's much clearer.